Update file names 411

3mma-mack opened the pull request:
Pull request to update output file names as part of issue #411. This includes updates to corresponding name constants and arguments in `prep_riboviz.nf`, `riboviz/workflow_r.py`, `riboviz/workflow_files.py`, and `generate_stats.figs.R`, as well as updating names in any documentations.

The file riboviz_outputs.md has been created to describe each of the output files in detail. 
Config file argument `do_pos_sp_nt_freq` has been updated to `output_metagene_normalized_profile`, with `riboviz/upgrade_config.py` updated to reflect this. Config files in example-datasets will likely need to be updated again to reflect this change.

The table of original and new names can be found [here.](https://github.com/riboviz/riboviz/issues/411#issuecomment-897720102)

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
 The workflow will summarise information about the number of reads in the input files and in the output files produced at each step of the workflow. This summary is produced by scanning input, temporary and output directories and counting the number of reads (sequences) processed by specific stages of a RiboViz workflow.
 
-The read counts file, `read_counts.tsv`, is written into the output directory.
+The read counts file, `read_counts_per_file_tsv`, is written into the output directory.
```
`read_counts_per_file_tsv` =>  `read_counts_per_file.tsv`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+Descriptions of specific files can be located quickly using the following links:
+
+**Output files for an entire run**
+* [TPMs_all_CDS_all_samples](#tpms_all_cds_all_samplestsv)
```
Change link text to 'TPMs_all_CDS_all_samples.tsv'

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+* [frame_proportions_per_ORF.pdf](#frame_proportions_per_ORFpdf)
+
+After a riboviz run, many output files are produced within the output directory.
+The output directory is specified by the parameter `dir_out` in `config.yaml`.
```
Replace `config.yaml` with 'the YAML configuration file.'

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+## `TPMs_all_CDS_all_samples.tsv` 
+
+A tsv file with transcripts per million (tpm) for all genes from successfully processed samples. This file is produced by the script `collate_TPMs.R`. This script uses the `ORF_TPMs_and_counts.tsv` file from each processed sample and lists the tmps of each sample for each gene, allowing for comparison between samples. 
```
tmps => tpms

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+Example of the top of the HTML file:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/html_output.jpg" alt="html_output image" width="500"/>
```
This is too brittle as if your repository goes the image link will break. Add the image to `docs/images/` then replace the URL with `<img src="../images/html_output.jpg" ... />`. Comment applies to all `<img>` links in this doc.

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+*Potential names*
+
+read_nt_compostion, nt_freq_read_length, read_nt_freq
```
Delete the "Potential names"

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+A meta feature plot showing the total number of reads present within a 75nt window around the start and stop codons. The reads occurring at each position for each gene are summed to give a total for each position, then plotted. It is expected to see a large peak just upstream of the start codon, due to ribosome binding being the slow step of translation. This is typically followed by regular repe ...
+
+Good quality 3nt periodicity plot:
```
Rename to 'Good quality meta feature plot'

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/3nt_periodicity.jpg" alt="Good quality 3nt_periodicity plot" width="500"/>
+
+3nt periodicity plot produced using incorrect annotation files:
```
Reword to 'Meta feature plot produced...'

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+A plot that shows the mean number of reads mapping to each position upstream and downstream of the start and stop codon for all genes. It is expected to have a peak at the start codon, with the majority of following positions having a relatively consistent mean that is comparatively low to the mean observed at the start codon. 
+
+Example pos_sp_rpf_norm_reads plot:
```
Reword to 'Example mean number of reads plot:'

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+## `ORF_TPMs_vs_features.tsv`
+
+A tsv file showing the tpm of a set of features and a value for that feature. The features are Length_log10 (log10 of the gene length); uATGs (number of upstream start codons); FE_atg (Free Energy at ATG); FE_cap (Free Energy at the cap); utr (length of the UTR); utr_gc (GC content of the UTR); and polyA (3' polyA tail on mRNA). This is produced by combining the tpms file with the features file,  ...
```
Define features as a bulleted list so they're easier to read.

mikej888 commented on docs/user/riboviz_outputs.md:
````diff
@@ -0,0 +1,396 @@
+YAL010C	4.82007633057095	Length_log10	2.69284691927723
+```
+
+Only output if `--features-file` was defined.
````
`--features-file` => `features_file`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/features.jpg" alt=" features plot" width="500"/>
+
+Only output if `--features-file` was defined.
```
`--features-file` => `features_file`

mikej888 commented on docs/user/riboviz_outputs.md:
````diff
@@ -0,0 +1,396 @@
+```
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
+
````
`--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+A tsv file showing the correlatation of different codons to features based on a provided tRNA file, which gives the Amino acids, the tRNA estimates, the tAI (tRNA Adaptation Index), Microarray values, and RNA.seq values for each codon. These are used to calculate mean ribosome-densities at the A/P/E sites for each codon.
+
+Produced by `generate_stats_figs` during step "Codon-specific ribosome densities for correlations with tRNAs" using functions `CalculateCodonSpecificRibosomeDensityTRNACorrelation` and `WriteCodonSpecificRibosomeDensityTRNACorrelation`.
```
`generate_stats_figs` => `generate_stats_figs.R`

mikej888 commented on docs/user/riboviz_outputs.md:
````diff
@@ -0,0 +1,396 @@
+
+```
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
````
`--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/codon_ribodens.jpg" alt="codon_ribodens plot" width="500"/>
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
`--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+## `metagene_start_ribogrid_by_length.pdf`
+
+A meta-feature heatmap showing the number of reads occurring at positions around the start codons of genes with the y axis as read length, and the colour intensity showing read count. It is expected that each length will show a peak of reads just upstream of the start codon for all lengths, shown by an intense dark purple, and then an observable 3nt periodicity following the peak, which will be m ...
```
`3nt_periodicity.tsv` => `metagene_start_stop_read_counts.tsv`

mikej888 commented on docs/user/riboviz_outputs.md:
````diff
@@ -0,0 +1,396 @@
+YAL064C-A	0	0	0	1	1	1
+```
+
+Only output if `--asite-disp-length-file` was defined.
````
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented on docs/user/riboviz_outputs.md:
````diff
@@ -0,0 +1,396 @@
+YAL058W	43	4	31	3.99564191533441e-08	0.175807992651532	4.21595642555769e-06
+```
+
+Only output if `--asite-disp-length-file` was defined.
````
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented on docs/user/riboviz_outputs.md:
```diff
@@ -0,0 +1,396 @@
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/3ntframe_propbygene.jpg" alt="3ntframe_propbygene plot" width="500"/>
+
+Only output if `--asite-disp-length-file` was defined.
```
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `nt_freq_per_read_position.tsv`
+* `metagene_normalized_profile_start_stop.pdf`
+* `metagene_normalized_profile_start_stop.tsv`
+* `ORF_TPMs_vs_features.tsv`: only output if `--features-file` was defined.
```
`--features-file` => `features_file`

mikej888 replied:
`--features-file` => `features_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `metagene_normalized_profile_start_stop.pdf`
+* `metagene_normalized_profile_start_stop.tsv`
+* `ORF_TPMs_vs_features.tsv`: only output if `--features-file` was defined.
+* `ORF_TPMs_vs_features.pdf`: only output if `--features-file` was defined.
```
`--features-file` => `features_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `ORF_TPMs_vs_features.pdf`: only output if `--features-file` was defined.
+* `ORF_TPMs_and_counts.tsv`
+* `normalized_density_APEsites_per_codon.tsv`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
+* `normalized_density_APEsites_per_codon.pdf`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
`--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `ORF_TPMs_and_counts.tsv`
+* `normalized_density_APEsites_per_codon.tsv`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
+* `normalized_density_APEsites_per_codon.pdf`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
+* `normalized_density_APEsites_per_codon_long.tsv` only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
`--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `normalized_density_APEsites_per_codon_long.tsv` only output if `--t-rna-file` and `--codon-positions-file` were defined.
+* `metagene_start_barplot_by_length.pdf`
+* `metagene_start_ribogrid_by_length.pdf`
+* `read_frame_per_ORF.tsv`: only output if `--asite-disp-length-file` was defined.
```
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `metagene_start_barplot_by_length.pdf`
+* `metagene_start_ribogrid_by_length.pdf`
+* `read_frame_per_ORF.tsv`: only output if `--asite-disp-length-file` was defined.
+* `read_frame_per_ORF_filtered.tsv`: only output if `--asite-disp-length-file` was defined.
```
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented on docs/user/prep-riboviz-operation.md:
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
+* `metagene_start_ribogrid_by_length.pdf`
+* `read_frame_per_ORF.tsv`: only output if `--asite-disp-length-file` was defined.
+* `read_frame_per_ORF_filtered.tsv`: only output if `--asite-disp-length-file` was defined.
+* `frame_proportions_per_ORF.pdf`: only output if `--asite-disp-length-file` was defined.
```
`--asite-disp-length-file` => `asite_disp_length_file`

mikej888 commented:
@3mma-mack I reran all the tests and all is still OK. I've re-reviewed all the changes and have suggested changes above. All of these relate to consistency in naming of parameters and files.

Two additional suggestions are:

* Add link `[riboviz output files and figures](./docs/user/riboviz_outputs.md)` to `README.md` in the Usage bullet list under "Running the RiboViz Nextflow workflow".
* Combine content from `prep-riboviz-operation.md` `## Read counts file` section into the relevant section of your `riboviz_outputs.md` page then delete this section from `prep-riboviz-operation.md`. Once deleted update `A [read counts file](#read-counts-file) ...` in `prep-riboviz-operation.md` to `A read counts file ...`. (so the docs describe the read counts file in one place only, `riboviz_outputs.md`).

3mma-mack commented:
Thanks for the feedback and suggestions! I'll make those changes ASAP

3mma-mack commented:
I have made the changes suggested by @mikej888, including changing references of file requirements from the format `--t-rna-file` to the format `t_rna_file`, rewording sentences where necessary and removing a few final occurrences of the old file names from `riboviz_outputs.md`.

I have also updated `riboviz_outputs.md` to include the information from the `## read counts` section of `prep-riboviz-operation.md` and added a link to `riboviz-outputs.md` to the README file.

mikej888 commented:
@3mma-mack, thanks. I just committed a change to fix a broken img URL. All looks fine to me for a merge into `develop`. @FlicAnderson do you have any comments?
